pool/sqrt_price_math.gno

```diff
@@ -55,49 +70,109 @@ func sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown(
+		quotient = (amount << 96) / liquidity
 	} else {
-		if amount <= consts.MAX_UINT160 {
-			quotient = (amount << 96) / liquidity
-		} else {
-			quotient = amount * consts.Q96 / liquidity
-		}
-		requireUnsigned(quotient, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown() || quotient(%d) >= 0", quotient))
-
-		require(sqrtPX96 > quotient, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown() || sqrtPX96(%d) must be greater than quotient(%d)", sqrtPX96, quotient))
-		return sqrtPX96 - quotient
+		quotient = amount * (consts.Q96 / liquidity)
 	}
-}
+
+	// quotient mutst be positive when amount and liquidity are positive
+
+	if add {
+		result := sqrtPX96 + quotient
+
+		require(
+			result >= sqrtPX96,
+			ufmt.Sprintf(
+				"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown() || expected result >= sqrtPX96, got: %d",
+				result,
+			),
+		)
+
+		return result
+	}
+
+	result := sqrtPX96 - quotient
+
+	require(
+		result >= 0,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown() || expected sqrtPX96 >= quotient, got: %d, %d",
+			sqrtPX96,
+			quotient,
+		),
+	)
+
+	return result
+}	
 
 func sqrtPriceMathGetNextSqrtPriceFromInput(
 	sqrtPX96 bigint,
 	liquidity bigint,
 	amountIn bigint,
 	zeroForOne bool,
 ) (sqrtQ bigint) {
-	requireUnsigned(sqrtPX96, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || sqrtPX96(%d) >= 0", sqrtPX96))
-	requireUnsigned(liquidity, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || liquidity(%d) >= 0", liquidity))
-	requireUnsigned(amountIn, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || amountIn(%d) >= 0", amountIn))
+	requireUnsigned(
+		sqrtPX96,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || expected sqrtPX96 >= 0. got: %d",
+			sqrtPX96,
+		))
+
+	requireUnsigned(
+		liquidity,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || expected liquidity >= 0. got: %d",
+			liquidity,
+	))
+
+	requireUnsigned(
+		amountIn,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || expected amountIn >= 0. got: %d",
+			amountIn,
+	))
+
 
 	if zeroForOne {
-		requireUnsigned(sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp(sqrtPX96, liquidity, amountIn, true), "[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || condition must return `v` >= 0__#1")
-		return sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp(sqrtPX96, liquidity, amountIn, true)
+		amount0Result := sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp(sqrtPX96, liquidity, amountIn, true)
+		requireUnsigned(
+			amount0Result,
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || condition must return `v` >= 0__#1",
```

Comment:
change 
```[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() || condition must return `v` >= 0__#1```

to

`[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromInput() #1 || amount0Result(%d) >= 0`